dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/DataRequestAlterationManager.kt

```diff
@@ -68,13 +69,39 @@ class DataRequestAlterationManager(
             messageRepository.saveAllAndFlush(dataRequestEntity.messageHistory)
             this.sendSingleDataRequestEmail(dataRequestEntity, contacts, message)
         }
-        if (requestStatus == RequestStatus.Answered) {
-            val correlationId = UUID.randomUUID().toString()
-            dataRequestedAnsweredEmailMessageSender.sendDataRequestedAnsweredEmail(dataRequestEntity, correlationId)
+        if (requestStatus != null) {
+            sendEmailBecauseOfStatusChanged(dataRequestEntity, requestStatus)
         }
         return dataRequestEntity.toStoredDataRequest()
     }
 
+    /**
+     * Method to send email if the status changed
+     * @param dataRequestEntity the id of the request entity
+     * @param status the patched request status
+     */
+    private fun sendEmailBecauseOfStatusChanged(
+        dataRequestEntity: DataRequestEntity,
+        status: RequestStatus,
+        correlationId: String = UUID.randomUUID().toString(),
+    ) {
+        when (status) {
+            RequestStatus.Answered -> {
+                dataRequestResponseEmailMessageSender.sendDataRequestResponseEmail(
+                    dataRequestEntity, TemplateEmailMessage.Type.DataRequestedAnswered, correlationId,
+                )
+            }
+            RequestStatus.Closed -> {
+                dataRequestResponseEmailMessageSender.sendDataRequestResponseEmail(
+                    dataRequestEntity, TemplateEmailMessage.Type.DataRequestClosed, correlationId,
+                )
+            }
+            else -> {
+                return
```

Comment:
i havent seen the other code yet, but i would expect an exception if status is something else.
The assurance that the function is not called with a different status should be handled outside of this function

Reply:
changed: throw IllegalArgumentException("Unable to send email. Unexpected status provided: $status")